src/world/World.h

```diff
@@ -37,17 +37,17 @@ class World {
 public:
     static constexpr const int MAX_SEARCH_RESULTS = 10;
 
-    static constexpr const int VISIT_AIRPORTS =    0b1;
-    static constexpr const int VISIT_AIRFIELDS =   0b10;
-    static constexpr const int VISIT_FIXES =       0b100;
-    static constexpr const int VISIT_NAVAIDS =     0b1000;
-    static constexpr const int VISIT_USER_FIXES =  0b10000;
-    static constexpr const int VISIT_EVERYTHING = (VISIT_AIRPORTS | VISIT_AIRFIELDS | VISIT_NAVAIDS | VISIT_FIXES | VISIT_USER_FIXES);
+    static constexpr const int VISIT_TOWERED_AIRPORTS = 0b1;
+    static constexpr const int VISIT_OTHER_AIRPORTS =   0b10;
+    static constexpr const int VISIT_FIXES =            0b100;
+    static constexpr const int VISIT_NAVAIDS =          0b1000;
+    static constexpr const int VISIT_USER_FIXES =       0b10000;
+    static constexpr const int VISIT_EVERYTHING = (VISIT_TOWERED_AIRPORTS | VISIT_OTHER_AIRPORTS | VISIT_NAVAIDS | VISIT_FIXES | VISIT_USER_FIXES);
 
-    using NodeAcceptor = std::function<void(const world::NavNode &node)>;
+    using NodeAcceptor = std::function<void(const world::NavNode *)>;
```

Comment:
Is there a reason to introduce raw pointers here and when passing the overlay helpers?

Reply:
My main thinking was for performance reasons when it is known that the pointer is only used for reference during the function call, and is not kept for persistent use or ongoing shared ownership. But the gains are probably only marginal, so I could convert back to shared pointers if it is preferred for style and extra safety.

(I seem to recall I did find one instance where a shared pointer actually led to objects not being destroyed at shutdown, due to circular shared pointer references, and in this case providing an interface as a raw pointer solved the issue, but it does mean that ownership, use and destruction order needs to be understood!)

Let me know if you want any of the raw pointer uses protected by wrapping as shared pointers.

Reply:
Understood. I think it's fine for the NavNodes, but things like the `LoadManager` or `IOverlayHelper` should be smart pointers. If shutdown is a problem, they could become `weak_ptr` if needed.

Reply:
I've just been reviewing the use of raw pointers for LoadManager and World, and it looks to me as if the same pattern is being used for these as is described above for the IOverlayHelper and NavNode pointers. (I could not find any issues related to order of destruction, so I suppose this was something I mis-remembered, or resolved at the time!)

The various xdata loader classes are instantiated with raw pointers to the LoadManager (via which they get a raw pointer to the World object). The loader classes are instantiated and destroyed by the LoadManager, so there is no danger of access after destruction (as above). However, the performance improvement of using raw pointers is probably not such an issue, and perhaps keeping to C++ smart pointer idiom is better.

I will convert these LoadManager and World raw pointer usage back to C++ smart pointers. Since this code is not actually part of the PR currently under discussion I will submit this as part of a new PR.

If you are satisfied with my justification for using raw pointers in the overlayed maps updates, then I think this PR could be accepted as is?